iceoryx2-bb/posix/tests/adaptive_wait_tests.rs

```diff
@@ -20,19 +20,48 @@ use std::time::Instant;
 const TIMEOUT: Duration = Duration::from_millis(50);
 
 #[test]
-fn adaptive_wait_wait_loop_initial_repetitions_plus_one_is_at_least_final_waiting_time() {
-    let start = Instant::now();
+fn adaptive_wait_wait_at_different_time_depends_on_repetition_times() {
```

Comment:
I have thought to test the `yield_now` behavior as well but haven't found a proper way. Defining 2 threads and testing the execution order is too fragile and i don't want to do so.

Reply:
In roughly a year, we wanted to add a testing platform to capture syscalls or inject failures. When this is available, I would such tests there.

I also think that there isn't too much value in verifying this at the moment. But later on it becomes important when we are aiming for 100% MC/DC coverage.